openedx/core/djangoapps/content_libraries/tests/test_content_libraries.py

```diff
@@ -287,10 +289,14 @@ def test_library_blocks(self):
         assert self._get_library(lib_id)['has_unpublished_changes'] is True
 
         # Publish the changes:
-        self._commit_library_changes(lib_id)
+        updated_date = datetime(2024, 6, 7, 8, 9, 10, tzinfo=timezone.utc)
+        with freeze_time(updated_date):
+            self._commit_library_changes(lib_id)
         assert self._get_library(lib_id)['has_unpublished_changes'] is False
         # And now the block information should also show that block has no unpublished changes:
         block_data["has_unpublished_changes"] = False
+        block_data["last_published"] = updated_date.isoformat().replace('+00:00', 'Z')
+        block_data["published_by"] = "Bob"
```

Comment:
Could you also test for the expected `last_draft_created` and `last_draft_created_by` data, both for when there's unpublished changes and when there isn't?

Reply:
Here: 3be2aad0390a34c86479567438495d501fc1bd0b